sycl/doc/extensions/KernelRHSAttributes/SYCL_INTEL_attribute_style.asciidoc

```diff
@@ -1,219 +0,0 @@
-making it clear to which kernel an attribute applies.
-
-A kernel attribute applied to the function as required by SYCL 1.2.1 looks like:
-
-[source,c++]
-----
-[[attrib]] void foo1() {};
-
-class f {
-  [[attrib]] void foo2() {};
-};
-----
-
-where `attrib` is a placeholder for any of the kernel attributes defined by the SYCL specification or extensions.
-
-This extension deprecates the SYCL 1.2.1 attribute style (attribute applied to
-a device function) and instead defines kernel attributes as attributes that apply to the
-function type.  The location of the resulting attributes looks like:
-
-[source,c++]
-----
-void bar1() [[attrib]] {};
-
-class f {
-  void bar2() [[attrib]] {};
-};
-
-class KernelFunctor {
-  public:
-  void operator()(sycl::item<1> item) [[attrib]] {};
-};
-
-auto bar3 = []()[[attrib]]{}; // Works on lambdas.  operator() type
-----
-
-The function type attributes have an effect when applied to a kernel function,
-do not propagate up or down call trees unless specified by a specific attribute,
-and the effect when applied to non-kernel functions or non-functions is implementation defined.
-
-== Modifications of SYCL 1.2.1 Specification
-
-=== Modify Section 6.7 (Attributes)
-
-==== Rename the section
-
-Rename Section 6.7 from "Attributes" to "Kernel attributes".
-
-==== Replace the entire section with:
-
-The SYCL programming interface defines attributes that augment the
-information available while generating the device code for a particular platform.
-The attributes in Table 1 are defined in the `sycl::` namespace
-and are applied to the function-type of kernel function declarations using
-{cpp}11 attribute specifier syntax.
-
-A given attribute-token shall appear at most once in each attribute-list.  The
-first declaration of a function shall specify an attribute if any declaration
-of that function specifies the same attribute.  If a function is declared with
-an attribute in one translation unit and the same function is declared without
-the same attribute in another translation unit, the program is ill-formed and
-no diagnostic is required.
-
-If there are any conflicts between different kernel attributes, then the behavior
-is undefined.  The attributes have an effect when applied to a device function and no
-effect otherwise (i.e. no effect on non-device functions and on anything other than a
-function).  If an attribute is applied to a device function that is not a kernel function
-(but that is potentially called from a kernel function), then the effect is implementation defined.
-It is implementation defined whether any diagnostic is produced when an attribute is applied
-to anything other than the function-type of a kernel function declaration.
-
-.Attributes supported by the SYCL programming interface
-[cols="2*"]
-|===
-|`reqd_work_group_size(dim0)`
-`reqd_work_group_size(dim0, dim1)`
-`reqd_work_group_size(dim0, dim1, dim2)`
```

Comment:
I'm not sure if they're already deprecated, but we might need to open a ticket to get the `intel::` variants of these deprecated in favor of the `sycl::` variants.

Reply:
This spec that I'm deleting is only about the *position* of the attributes.  The `intel::` namespace variant of `reqd_work_group_size` attribute is defined in ["sycl/doc/extensions/ReqdWorkGroupSize/SYCL_INTEL_reqd_work_group_size.asciidoc"](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/ReqdWorkGroupSize/SYCL_INTEL_reqd_work_group_size.asciidoc), which is **not** removed in this PR.  I was thinking of keeping that extension and moving it to the "deprecated" folder, which I will do in a separate PR.

I'm also not sure if the compiler issues a "deprecated" warning for the old form of this attribute.  I'll check and create an issue if it does not.

Reply:
Makes sense, thanks.  I was confused because we seem to have documented these extensions in a few places...  Thanks again for leading the charge on tidying this stuff up.

Reply:
It turns out that the compiler does not even recognize the old form of this attribute in the `[[intel::]` namespace, so there is no need to keep that spec even in the "deprecated" folder.  I created a separate PR to remove that spec in #5387.